src/parser/mod.rs

```diff
@@ -5434,6 +5436,13 @@ impl<'a> Parser<'a> {
                 }
             }
 
+            let limit_by =
+                if dialect_of!(self is ClickHouseDialect) && self.parse_keyword(Keyword::BY) {
```

Comment:
in general I think we try to have `GenericDialect` support the union of all sql dialects, if possible. So this might be something like

```suggestion
                if dialect_of!(self is ClickHouseDialect | GenericDialect) && self.parse_keyword(Keyword::BY) {
```